addons/website_sale_coupon/controllers/main.py

```diff
@@ -16,6 +17,27 @@ def pricelist(self, promo, **post):
             request.session['error_promo_code'] = coupon_status['error']
         return request.redirect(post.get('r', '/shop/cart'))
 
+    @http.route(['/shop/cart/update_json'], type='json', auth="public", methods=['POST'], website=True, csrf=False)
+    def cart_update_json(self, product_id, line_id=None, add_qty=None, set_qty=None, display=True, **kw):
+        if line_id and product_id and set_qty == 0:
+            line = request.env['sale.order.line'].sudo().browse(int(line_id))
+            if line.is_reward_line:
+                status = request.session.get('pending_coupon') or {}
+                promo_code = status.get('promo_code')
+
+                if promo_code:
+                    pending_coupon = request.env['coupon.coupon'].sudo().search([('code', '=', promo_code)], limit=1)
```

Comment:
```suggestion
                    pending_coupon_sudo = request.env['coupon.coupon'].sudo().search([('code', '=', promo_code)], limit=1)
```
please when a variable hold a sudo record, name it with sudo, it will help people reading the code and especially help avoiding security issue if the person modifying your code doesn't realize it is sudo